From 8ec9b36ea829630b66a10ecbe4fe736648a65860 Mon Sep 17 00:00:00 2001
From: Davide Cavalca <davide@cavalca.name>
Date: Sat, 11 Jul 2020 15:24:39 -0700
Subject: [PATCH] fix subvolume umount path

---
 appcreate/partitionedfs.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/appcreate/partitionedfs.py b/appcreate/partitionedfs.py
index 86e709c..a1597ea 100644
--- a/appcreate/partitionedfs.py
+++ b/appcreate/partitionedfs.py
@@ -256,7 +256,7 @@ class PartitionedMount(Mount):
             ordered += others
              
             for s in ordered:
-                subprocess.call(['umount', s['mountpoint']])
+                subprocess.call(['umount', "%s%s" % (self.mountdir, s['mountpoint'])])
     
         for mp in self.unmountOrder:
             if mp == 'swap':
-- 
2.26.2

