From 4004c87d935712cdafe0e1319387d55e651b7703 Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa13@gmail.com>
Date: Fri, 2 Dec 2016 08:52:11 -0500
Subject: [PATCH] Fix import for compatibility with livecd-tools v24

---
 appcreate/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/appcreate/__init__.py b/appcreate/__init__.py
index 4bed653..0c4cebf 100644
--- a/appcreate/__init__.py
+++ b/appcreate/__init__.py
@@ -19,7 +19,7 @@
 
 from imgcreate.live import *
 from imgcreate.creator import *
-from imgcreate.yuminst import *
+from imgcreate.dnfinst import *
 from imgcreate.kickstart import *
 from imgcreate.fs import *
 from imgcreate.debug import *
-- 
2.9.3 (Apple Git-75)

