From a0feeeb6630ecc08c407255e15fa9dfbf00e1c44 Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Sun, 12 May 2013 15:58:48 -0500
Subject: [PATCH] couple of typo fixes in extlinux support from mattdm

---
 appcreate/appliance.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/appcreate/appliance.py b/appcreate/appliance.py
index 54ac376..dbc66d7 100644
--- a/appcreate/appliance.py
+++ b/appcreate/appliance.py
@@ -307,9 +307,9 @@ class ApplianceImageCreator(ImageCreator):
         extlinux += "ui menu.c32\n"
         extlinux += "menu autoboot Welcome to %s. Automatic boot in # second{,s}. Press a key for options.\n" % (self.name)
         extlinux += "menu title %s Boot Options.\n" % (self.name)
-        extlinux += "menu hiddenmenu\n"
+        extlinux += "menu hidden\n"
         extlinux += "timeout 1\n"
-        extlinux += "totaltimeout 60\n\n"
+        extlinux += "totaltimeout 600"
 
         versions = []
         kernels = self._get_kernel_versions()
-- 
1.8.2.1

